File: frontend/src/services/productService.js
import api from './api'

export const productService = {
  async getAll(search = '') {
    const response = await api.get('/products', { params: { search } })
    return response.data
  },

  async getAllWithQuantity(search = '') {
    const response = await api.get('/products', { params: { search, with_quantity: '1' } })
    return response.data
  },

  async getById(id) {
    const response = await api.get(`/products/${id}`)
    return response.data
  },

  async create(product) {
    const response = await api.post('/products', product)
    return response.data
  },

  async update(id, product) {
    const response = await api.put(`/products/${id}`, product)
    return response.data
  },

  async delete(id) {
    const response = await api.delete(`/products/${id}`)
    return response.data
  }
}

File: backend/services/product_service.go
package services

import (
	"pharmacy-erp/models"
	"pharmacy-erp/repository"
)

type ProductService struct {
	repo *repository.ProductRepository
}

func NewProductService() *ProductService {
	return &ProductService{repo: repository.NewProductRepository()}
}

func (s *ProductService) Create(p *models.Product) error {
	return s.repo.Create(p)
}

func (s *ProductService) GetAll(search string) ([]models.Product, error) {
	return s.repo.FindAll(search)
}

func (s *ProductService) GetAllWithQuantity(search string) ([]repository.ProductWithQuantity, error) {
	return s.repo.FindAllWithQuantity(search)
}

func (s *ProductService) GetByID(id uint) (*models.Product, error) {
	return s.repo.FindByID(id)
}

func (s *ProductService) Update(p *models.Product) error {
	return s.repo.Update(p)
}

func (s *ProductService) Delete(id uint) error {
	return s.repo.Delete(id)
}
